src/OnboardingSPA/pages/Steps/DesignColors/index.js

```diff
@@ -41,15 +41,21 @@ const StepDesignColors = () => {
 	}, [] );
 
 	const getStylesAndPatterns = async () => {
-		const pattern = await getPatterns( currentStep.patternId, true );
-		if ( pattern?.error ) {
+		const designColorPatterns = await getPatterns(
+			currentStep.patternId,
+			true,
+			currentData?.data?.partHeader ? currentData?.data?.partHeader.split( '/' )[ 1 ] : '',
+		);
```

Comment:
Like we discussed the other day, it seems unnecessary to duplicate this logic in every single step (it can cause misses and also make changing this code a tedious process), we need to consolidate this using either of the approaches mentioned in my comment regarding removing `headerMenuSlug`.

Reply:
Removed sending partHeader from the frontend ... it is now being read from the DB